src/lib/entropy/rdseed/rdseed.cpp

```diff
@@ -36,13 +39,17 @@ size_t Intel_Rdseed::poll(RandomNumberGenerator& rng)
 #endif
             if(1 == cf)
                {
-               rng.add_entropy_T(r);
+               seed.push_back(r);
                break;
                }
             }
          }
+
+      rng.add_entropy(reinterpret_cast<const uint8_t*>(seed.data()),
```

Comment:
Check here if all `_rdseed32_step` failed and if `seed` is zero?

Reply:
That shouldn't happen since we only append to the seed vector if cf == 1?

Reply:
What happens if cf != 0 for every loop iteration so that seed is empty and then add_entropy is called on the empty seed?

Reply:
It will cause a HMAC_DRBG update sequence which is useless but harmless. I added a guard for seed.size() > 0.